frontend/src/main/scala/bloop/engine/caches/SemanticDBCache.scala

```diff
@@ -57,6 +65,44 @@ object SemanticDBCache {
     }
   }
 
+  private def fallbackSemanticdbFetch(
+      scalaVersion: String,
+      artifact: DependencyResolution.Artifact,
+      logger: Logger
+  ): Either[String, AbsolutePath] = {
+    // if scala version is no longer supported find the latest supported semanticdb version
+    def versionFuture =
+      Task {
+        coursierapi.Complete
+          .create()
+          .withScalaVersion(scalaVersion)
+          .withScalaBinaryVersion(scalaVersion.split('.').take(2).mkString("."))
+          .withInput(s"org.scalameta:semanticdb-scalac_$scalaVersion:")
+          .complete()
+          .getCompletions()
+          .asScala
+          .lastOption
+      }
```

Comment:
nitpick - `versionFuture` returns Task despite its name. Moreover, you're instantly converting it to future via `versionFuture.runAsync(ExecutionContext.ioScheduler)`, maybe you could go to the Future directly? There is no point in using Task in this case.

Reply:
Sure, I just wanted to be consistent, but maybe it doesn't really make sense. Changed!